recipes/libhal/all/conanfile.py

```diff
@@ -50,28 +58,37 @@ def lazy_lt_semver(v1, v2):
         compiler = str(self.settings.compiler)
         version = str(self.settings.compiler.version)
         minimum_version = self._compilers_minimum_version.get(compiler, False)
-
         if minimum_version and lazy_lt_semver(version, minimum_version):
             raise ConanInvalidConfiguration(
                 f"{self.name} {self.version} requires C++{self._min_cppstd}, which your compiler ({compiler}-{version}) does not support")
 
-    def layout(self):
-        basic_layout(self)
-
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-            destination=self.source_folder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def build(self):
+        if Version(self.version) == "0.0.0":
+            replace_in_file(
+                self, 
+                os.path.join(self.source_folder, "include", "libhal", "percentage.hpp"), 
```

Comment:
I can't find any details in the PR description about the need for this patch - is there a compiler error on some specific compiler/version? 

On the other hand, I can see that version `0.0.0` is described as pre-alpha, perhaps we can consider removing the version altogether from the recipe instead.

Reply:
It looks like the spaceship operator is supposed to return an `int` and not a `bool`: https://en.cppreference.com/w/cpp/language/operator_comparison#Three-way_comparison

A patch file for this would be nice.

Reply:
@jcar87 
As jwillikers adviced, I create a patch file with patch_desscription.
Please review again.